### PR TITLE
[mini]For RZ, added assert that periodicity is not set radially

### PR DESCRIPTION
**Pull request by dpgrote**
This will abort if the user has periodicity turned on in the radial direction.

### CODE REVIEW

File: Source/WarpX.cpp [hunk 1]
```diff
@@ -554,6 +554,11 @@ WarpX::ReadParameters ()
 
         // Only needs to be set with WARPX_DIM_RZ, otherwise defaults to 1
         pp.query("n_rz_azimuthal_modes", n_rz_azimuthal_modes);
+
+#if defined WARPX_DIM_RZ
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(Geom(0).isPeriodic(0) == 0,
```
Comment (LDAmorim) [inline Source/WarpX.cpp:559]: Minor question, do we need to indicate that `Geom()` is from `GetInstance()`?
```suggestion
        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(GetInstance().Geom(0).isPeriodic(0) == 0,
```

**Review by LDAmorim — APPROVED** [review]
Thank you very much for this super quick fix @dpgrote !!

File: Source/WarpX.cpp (hunk 1 above)
Comment (dpgrote) [reply Source/WarpX.cpp:559]: Good question. Geom is used earlier in this routine and I just copied that code. 'ReadParameters' is not a static routine so it has direct access to the attributes of the WarpX object. Compare this to `CellSize` which is declared static.

Comment (LDAmorim) [reply Source/WarpX.cpp:559]: >  'ReadParameters' is not a static routine so it has direct access to the attributes of the WarpX object.
Thanks for explaining @dpgrote ! That does makes sense.

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**